File: 数据结构与算法/排序/code_java/bubbleSort.java
import java.util.Scanner;

//右侧为排好序的一侧
public class bubbleSort {
    public void swap(int[] arr,int i,int j){
        int tmp=arr[i];
        arr[i]=arr[j];
        arr[j]=tmp;
    }
    public int[] sortFunc(int[] arr){
        boolean isSwap=false;
        for (int i=arr.length-1;i>=0;i--){
            int j=0;
            while(j<i){
                if(arr[j]>arr[j+1]){
                    swap(arr,j,j+1);
                    isSwap=true;
                }
                j=j+1;
            }
            if(isSwap=false)
                break;
        }
      return arr;
    }

}

File: 数据结构与算法/排序/code_pyhon/bubble_sort.py
def bubble_sort(l):
    r=list(range(len(l)))[::-1]
    for i in r:
        is_swap=False
        j=0
        while j<i:
            if l[j]>l[j+1]:
                l[j],l[j+1]=l[j+1],l[j]
                is_swap=True
            j=j+1
    return l

def main():
    l=[1,6,3,5,9,2]
    l=bubble_sort(l)
    print(l)

if __name__=='__main__':
    main()
